### PR TITLE
php artisan backpack:field command

### PR DESCRIPTION
**Pull request by karandatwani92**
#73 
New Feature:
What I did:
Created an artisan command to create a custom field really quick (with a stub).

What happens:
Run `php artisan  backpack:field my_field` and it would copy `text` field definition and create a field with new given name.

Run `php artisan  backpack:field my_field --withAssets` and it would copy `text` field definition and append `with-assets` Stub to it.

Run `php artisan  backpack:field my_field --from=upload` and it would copy `from` field definition

### CODE REVIEW

**Comment by pxpm** [comment]
I still think that we should have "empty field stubs" to show how the field structure is organized and the "necessary" things to make it work when building a field from scratch. 

I would only publish text field if users use `--from=text` . 

Also notice that when using `--from`, you should not be able to use `--withAssets` or any other option that would work for "empty stubs" but can break actual fields.

**Review by tabacitu — CHANGES_REQUESTED** [review]
Good job @karandatwani92 . Let's incorporate Pedro's feedback too. I've rephrased it here with actionable steps. Also, found a small bug.

Good job 👏👏👏

File: src/Console/Commands/FieldBackpackCommand.php [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+     * @return string
+     */
+    protected function getStub()
+    {
+        $field = "text";
+        if ($this->option('from')) {
+            $field = Str::of($this->option('from'));
+        }
+
+        return base_path("vendor/backpack/crud/src/resources/views/crud/fields/$field.blade.php");
```
[the remark this answers is not in the document]
Comment (pxpm) [reply src/Console/Commands/FieldBackpackCommand.php:51]: This should be solved 100%. 

Pro pushes the packages view namespaces in the class @tabacitu mentioned. 

We should to try to match the field with the namespaces available in ViewNamespaces class.

File: src/Console/stubs/with-assets.stub [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+@push('crud_fields_scripts')
+  @loadOnce(asset('myFieldScript.js')
+
+@loadOnce('bpFieldInitMyFieldElement')
+    function bpFieldInitMyFieldElement(element) {
+    // element is the jquery wrapped element where init function is defined. 
+    }
+@endLoadOnce
+
+@endpush
```
[the remark this answers is not in the document]
Comment (pxpm) [reply src/Console/stubs/with-assets.stub:19]: Totally this is what I meant. 
I agree that `--withAssets` could be removed .
If you dont need assets you probably just need a text field and publish it with --from=text

Cheers

[the remark this answers is not in the document]
Comment (tabacitu) [reply src/Console/stubs/with-assets.stub:19]: Whoohoo

vibing-jojo-fletcher-gif-by-cnbc-prime

File: src/Console/Commands/FieldBackpackCommand.php [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        $path = $this->getPath($name);
+
+        if ($this->alreadyExists($this->getNameInput())) {
+            $this->error("Error : $this->type already existed!");
+
+            return false;
+        }
+
+        $this->infoBlock("Creating {$name->replace('_', ' ')->title()} {$this->type}");
+        $this->progressBlock("Creating view <fg=blue>resources/views/vendor/backpack/crud/fields/{$name->snake('_')}.blade.php</>");
```
[the remark this answers is not in the document]
Comment (promatik) [reply src/Console/Commands/FieldBackpackCommand.php:82]: @tabacitu that is my issue 😅
The progressBlock helper uses the last 20 chars for the info message on the right, I'll fix it soon 👌

File: src/Console/Commands/FieldBackpackCommand.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (karandatwani92) [reply src/Console/Commands/FieldBackpackCommand.php]: @tabacitu Done, please check.

File: src/Console/stubs/with-assets.stub (hunk 2 above)
[the remark this answers is not in the document]
Comment (karandatwani92) [reply src/Console/stubs/with-assets.stub]: @tabacitu Done, please check.

**Comment by tabacitu** [comment]
Excellent! Well done @karandatwani92 . The only change I've done was to properly indent the stub file, clean it up and add a few more comments. The easier we make it for devs to understand how things are done inside that file, the better.

File: src/Console/Commands/FieldBackpackCommand.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (promatik) [reply src/Console/Commands/FieldBackpackCommand.php:98]: https://github.com/Laravel-Backpack/CRUD/pull/4656 👌

**Pull request merged**